rust/src/ldap/ldap.rs

```diff
@@ -245,6 +347,140 @@ impl LdapState {
 
         return AppLayerResult::ok();
     }
+
+    fn parse_request_udp(
+        &mut self, flow: *const Flow, stream_slice: StreamSlice,
+    ) -> AppLayerResult {
+        let input = stream_slice.as_slice();
+        let _pdu = Frame::new(
+            flow,
+            &stream_slice,
+            input,
+            input.len() as i64,
+            LdapFrameType::Pdu as u8,
+            None,
+        );
+        SCLogDebug!("ts: pdu {:?}", self.request_frame);
+
+        match ldap_parse_msg(input) {
```

Comment:
Should not this code be factorized ? even if udp will still  return AppLayerResult::err(); on incomplete